### PR TITLE
add token authentication for createUser mutation

### PR DESCRIPTION
**Pull request by JoaoArroyo004**
- Add token authentication to createUser mutation
- Add test cases in which the token sent to createMutation is invalid or empty.

### CODE REVIEW

File: src/authenticate.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { DecodedToken } from './interfaces';
+import { CustomError } from './custom-error';
+import { verify } from 'jsonwebtoken';
+
+export function authenticate(token: string) {
+  const hasToken = !!token;
+  if (!hasToken) {
```
Comment (alanraso) [inline src/authenticate.ts]:
```suggestion
  if (!token) {
```

Se o token for alguma coisa que avalia pra false no javascript (undefined, null, string vazia, ...), ele já entraria no if e jogaria o erro.

File: src/test/create-user.test.ts [hunk 2]
```diff
@@ -16,14 +18,57 @@ describe('Create User Mutation', () => {
...
+
+  it('should not authenticate due to token generated with incorrect key', async () => {
+    const newUser = {
+      password: 'example_user',
+      email: 'first@example.com',
+      birthDate: '12/09/2004',
+      name: 'user 1',
+    };
+
+    const token = generateInvalidToken({ email: newUser.email, id: '1' }, false);
```
Comment (matheusflds) [inline src/test/create-user.test.ts]: nit: o seu `generateToken` podia ter um parametro opcional para a chave, e ai vc mandaria a chave errada ao invés de criar esse método generateInvalidToken direto no codigo "principal", o que acaba poluindo a leitura

File: src/test/helper.ts [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
         }
       }`,
-    variables: variables,
-  });
+      variables: variables,
+    },
+    {
+      headers: {
+        Authorization: `Bearer ${token}`,
+        'Content-Type': 'application/json',
```
Comment (alanraso) [inline src/test/helper.ts]: Dava algum erro sem esse header? Se não me engano é o padrão do axios.

**Review by matheusflds — APPROVED** [review]

File: src/token-generator.ts [hunk 4]
```diff
@@ -6,3 +6,10 @@ export const generateToken = (tokenParams: { email: string; id: string }, rememb
   const token: string = sign(tokenParams, secretKey, tokenOptions);
   return token;
 };
+
+export const generateInvalidToken = (tokenParams: { email: string; id: string }, rememberMe: boolean): string => {
+  const wrongKey = 'incorrectKey';
+  const tokenOptions = { expiresIn: rememberMe ? '7d' : '10m' };
+  const token: string = sign(tokenParams, wrongKey, tokenOptions);
+  return token;
+};
```
Comment (alanraso) [inline src/token-generator.ts]: No geral é uma boa prática separar em lugares diferentes os códigos que são usados apenas pra teste, e os que são usados na aplicação. Isso porque quando fomros gerar o código transpilado do servidor, nós vamos querer excluir todos os códigos dos testes de integração para termos um bundle menor. Então essa boa prática nos ajuda nisso já. Faz sentido?

Nos projetos a gente costuma deixar todos esses utilitários de teste em algum lugar dentro da pasta `test`

**Review by alanraso — APPROVED** [review]

File: src/test/helper.ts (hunk 3 above)
Comment (JoaoArroyo004) [reply src/test/helper.ts]: Realmente, é desnecessário explicitamente adicioná-lo. Obrigado!

File: src/token-generator.ts (hunk 4 above)
Comment (JoaoArroyo004) [reply src/token-generator.ts]: Por ser exatamente a mesma função do generateToken, mudando apenas a chave para uma incorreta, implementei o que o Matheus pontuou. Acredito que isso também resolva este ponto.
